### ISSUE TITLE
Doesn't recurse all the way through a complex object graph

### ISSUE THREAD

**Issue by sjmcdowall** [tomekwi/map-object-recursive.js#1]
I tried use this with camelcase-keys and noticed two problems:

1 --It doesn't call the function on the current key if the value is an object 

2 -- if the value is an array of objects it also doesn't recurse (or so it appears) 

I fixed issue #1 by changing the code block to be like 

```
mapObjectRecursive._mapOrRecurse = function mapOrRecurse(mappingFunction) {
    return function mappingOrRecursing(key, value, object) {
        // If `value` is a plain object, recurse.
        if (typeof value === 'object' &&
            !(value instanceof Array)) {
            var result = mappingFunction(key, value, object);
            return [result[0], mapObj(result[1], mappingOrRecursing)];
        }
```

But #2 still exists .. looking at that ..

**Comment by tomek-he-him** [comment]
Hey @sjmcdowall, thanks a lot for the bug report! Would you be interested in taking over maintenance of this module?

**Comment by sjmcdowall** [comment]
Thanks -- but maybe you should direct people to "map-object" and their 4.0 (?) version that now has a "deep:true" option that recurses (and does it nicely).. ???

**Comment by tomek-he-him** [comment]
Good idea, thanks! https://github.com/tomekwi/map-object-recursive.js/commit/bd0c8c0343e61017093a262cf11c27b245e875e2

**Comment by tomek-he-him** [comment]
Actually, I’ll keep this open to let people know what’s wrong with the implementation.
